### ISSUE TITLE
Review Dialect: Hindi (Python)

### ISSUE THREAD

**Issue by madiedgar** [legesher/legesher-translations#121]
## Review Hindi Language Translations for the Python Programming Language

In order to review this language to the project, complete the steps detailed in the [translation guide](https://docs.legesher.io/legesher-translations/translation-guide) (from Step # 6 onward). We are primarily working developing Legesher for the Python programming language. In the future we hope to add more programming languages.

**Reviewing** a translation is extremely important as we want to best understand the culture and context behind the words we select as keywords for the programming languages. The more perspectives the better! 👀Please take your time in being thorough with these translations. Think back when you were learning how to program and how context of words help you bridge together meaning and functionality.

> ⚠️ Note: this issue is to review the Python keywords and builtin functions only. Any additional translation reviews are encouraged, but not necessary.

**To complete this issue**:
- [x] Update any content in this issue from `Hindi` to the name of the language. (Romanian, Arabic, Welsh etc.)
- [x] Title this issue "Review Language: Hindi (Python)"
- [x] Update the Python section of the Hindi's `locale/__.yml` file (files are named using the language abbreviation) by adding comments as followed:
  - Leave a `✅`after a translation if you agree that's the right word to give context to the keyword
  - Leave a `❓` after a translation if you think there may be a better translation / form of the translation
  - Leave a `🤔` after a word that you're not sure if there is a translation
- [x] Merge changes into branch titled `hindi-translation` on this repo. Please reference this issue in the pull request.

### Reviewers Needed

In order to add the translations for official use across Legesher, the translations will need to be approved and accepted by __5 additional native / fluent speakers__. If you would like to be one of the reviewers for this language, please mention in this issue! :smile: Also feel free to loop in a friend that might be able to help with the translations! Please refer to the [translation guide](https://docs.legesher.io/guide/contributor/translation-guide) for the next step in the review process.

**Review Status:**
- [x] `Add New Dialect` Issue (#91)
- [x] Initiator: @parammittal16 (#114, [commit details](https://github.com/legesher/legesher-translations/commit/b7bdfde0300221875df7ecc7b43ebea18a28a4ea))
- [x] Reviewer 1: @iofall (#126)
- [x] Reviewer 2: @sshiv5768 (#135, #148, #151)
- [x] Reviewer 3: @Ln11211 (#251)
- [x] Reviewer 4: @nik132-eng (#262)
- [x] Reviewer 5: @aadii0408 (#268)
- [ ] Reviewer 6: @sneha-thyagarajan 

Thank you to @Nidhir2k1, @bhaveshgoyal182, @179priyasoni for your reviews in other programming languages.

**Comment by cedarfall** [comment]
I could help as Reviewer 1 for this. @madiedgar reviewers just have to add a comment with ✅, ❓, 🤔 depending on the translation after every word in the `locale/hi.yml` file in the `hindi-translation` branch right?

**Comment by madiedgar** [comment]
Yes exactly @cedarfall! If you think there is a better translation then please mention it. 😄 Excited to see your contribution! 👍

**Comment by nagendrady** [comment]
hi,iam new to open source development .but iam intrested to code for this.can u help with this.

**Comment by cedarfall** [comment]
@nagendrady yes sure, we will help you out. Feel free to comment here on any doubts you may have.

**Comment by nagendrady** [comment]
@cedarfall @nagendrady can i any other editor other than Atom like visual studio code
?

**Comment by cedarfall** [comment]
Yes @nagendrady you can use any editor. Also since this is a simple translation issue, you can also just edit the files on GitHub itself.

**Comment by nagendrady** [comment]
@cedarfall how can i add words for telugu?

**Comment by cedarfall** [comment]
@nagendrady this issue is for reviewing the Hindi translation. You can open a new issue for adding the Telugu language by following the steps as detailed in [the translation guide](https://legesher.readme.io/docs/translation-guide).

**Comment by madiedgar** [comment]
Thanks @cedarfall for all your help! Definitely great resources to check out 😄 

Hey @nagendrady! We can also set up a branch to get Telugu added to the list of languages 👍 I'll mention you on the new issues.

**Comment by madiedgar** [comment]
Check out this issue @sshiv5768

**Comment by sshiv5768** [comment]
I would love to do this task @madiedgar

**Comment by sshiv5768** [comment]
Hey, @madiedgar  I have added some more translations,please checkout them.

**Comment by mihirg008** [comment]
I could help as Reviewer 3 for this. Can you help me with the process? @madiedgar @sshiv5768

**Comment by madiedgar** [comment]
That would be wonderful @mihirg008 👍 Happy to help! Here's some helpful tips to get you started, but feel free to reach out if you have more questions! 

1️⃣ First tip is to work off the `hindi-translation` branch for your changes.

2️⃣ When reviewing a translation, It is important to keep all of the iterations of review in tact, so we're able to refer to them for the final review. Here are a couple of working examples of the key in the issue description.

**Line 1289-1290** would look something like this:
```yml
abs: "एब्स" # ❓ "परम", @you "परम"
all: "सभी" # ❓ "सब", @you "सब"
```

For something like **line 1299** where the English keyword is still in place (`chr: "chr"`), this would be acceptable:
```yml
chr: "चर" # ❓ "अक्षर" I believe data types should be kept as it is just change them phonetically if needed, @you ✅
```

Also, if you agree with the translation how it is, add a ✅  so we know its great! So like for **line 1297**:
```yml
bytes: "बाइट्स" # ✅✅
```

_You don't have to add your name by each of your translation reviews, but it can be helpful if you leave a comment/explanation as to why you feel a word better describes the keyword or you have additional ideas._

**Comment by sshiv5768** [comment]
Hey, @madiedgar. I want to contribute more in this project, will guide me through it?

**Comment by Nidhir2k1** [comment]
Would love to review :)

**Comment by madiedgar** [comment]
@Nidhir2k1 we'd love that! 👍 Assigned you to the issue. Again, if you have any questions, just reach out.

**Comment by bhaveshgoyal182** [comment]
Hey @madiedgar ,
I would love to review it too :)

**Comment by madiedgar** [comment]
@bhaveshgoyal182 That would be great! I'll assign you!

**Comment by 136tejas** [comment]
hey @madiedgar i would like to review it.

**Comment by madiedgar** [comment]
Thanks @136tejas! I just assigned you 👍

**Comment by 179priyasoni** [comment]
HEY @madiedgar  I would like to review .

**Comment by madiedgar** [comment]
Great @179priyasoni, assigning you now!

**Comment by sshiv5768** [comment]
@madiedgar, i added some more reviews in Hindi translations but this time i followed your instructions of changing branch. Please check it

**Comment by Ln11211** [comment]
@madiedgar , can I work on this ?, are there any more files to translate or review ?

**Comment by madiedgar** [comment]
Absolutely @Ln11211 - you'll want to follow the [translation guide](https://docs.legesher.io/legesher-translations/translation-guide) from Step #6 because the initial review has already been made. 

Tip: when you fork this repo, you'll want to make your changes on the `hindi-translation` branch of your fork and merge them into this repo's `hindi-translation` branch. You'll be specifically working on this file for this issue: https://github.com/legesher/legesher-translations/blob/hindi-translation/locale/hi.yml#L1614

**Comment by Ln11211** [comment]
@madiedgar , Can I start with translating JAVA keywords ? or is this Issue only for reviewing and translating Python keywords ?(But most of the keywords seems to have a good translation and are already reviewed) except few which have the ❓ and 🤔, so should I work on them specifically ?

**Comment by Ln11211** [comment]
@madiedgar , I was wondering, Have you also set up a "Telugu" translation branch as @nagendrady suggested ?

**Comment by madiedgar** [comment]
Great questions @Ln11211 👍 In order for a translation to be officially accepted it needs to be accepted and approved by 5 additional native speakers. Each keyword translation needs to have at least 6 individuals' input, for example: `bytes: "बाइट्स" # ✅✅✅✅✅`. Your input on the translations with ❓ and 🤔 would be extremely helpful. 

After you review the Python keywords, I'm more than happy to accept more PR's for other programming languages. Python is the first programming language we are rolling out. 

Yes- that translation is also set up and ready for reviews! See this issue: #172 
https://github.com/legesher/legesher-translations/tree/telugu-translation

**Comment by Ln11211** [comment]
@madiedgar One last doubt, so If we find a better word for a keywords marked with ❓, are we allowed to change(the existing translation) and write a new one followed with a comment(reasoning why is it better) ? And is it necessary to mention our @ after a ✅(like ✅ @Ln11211) ?

And yes, I am also planning to work on Telugu only after I am done with Hindi..
@madiedgar Thanks a lot for helping me begin with my work.....

**Comment by madiedgar** [comment]
These are great questions @Ln11211, and it helps me be clearer when I'm writing documentation - so thank you!

> If we find a better word for keywords marked with ❓, are we allowed to change(the existing translation) and write a new one followed with a comment(reasoning why is it better) ?

If the previous reviewers marked a translation with a ❓, then any insight you can provide for that word would be helpful. Add your review in the comments with your username so we know who to ask if we have any clarifying questions. The order of the comments will be helpful down the road when we are deciding on translations. So if you are reviewing a translation, keep everything in the comments, only the initial reviewers will input in the quotations. For example:
```yml
bytes: "" # ❓, ❓, "बाइट्स" because __________ @Ln11211 
```

> is it necessary to mention our @ after a ✅(like ✅ @Ln11211) ?

No, not necessary. If you'd like to add any specific commentary with your ✅, then please add your username. 👍 

> And yes, I am also planning to work on Telugu only after I am done with Hindi..

That would be wonderful!

**Comment by Ln11211** [comment]
@madiedgar , How many times have the "built in functions" of python Hindi file been reviewed ? because I suspect that someone reviewed it only once but used two two keys( ✅ &❓ or 🤔 & ✅ ) together, is it allowed to use two keys at a time ?
Like for instance in  [line 1289](https://github.com/legesher/legesher-translations/blob/hindi-translation/locale/hi.yml#L1289), [line 1303](https://github.com/legesher/legesher-translations/blob/hindi-translation/locale/hi.yml#L1303).

![image](https://user-images.githubusercontent.com/91385411/137589399-675cb10b-1b46-4ab4-8322-2caa573d039e.png)

**Comment by Ln11211** [comment]
@madiedgar , I have successfully reviewed all of the python translations thoroughly, can I raise a pr ?

**Comment by madiedgar** [comment]
Happy October Team!! We're so close to move Hindi to an official language in Legesher! We need at least one more person to review Python's keywords and builtin functions in order to move it to the next stage in the review process. 

Also - if you have already contributed and see 👀 next to your username in `locale/hi.yaml`, please update the translation with your own commentary ✅  / ❓ /  🤔 . 

@mihirg008
@sshiv5768
@iofall
@bhaveshgoyal182
@nik132-eng
@136tejas
@aadii0408
@Nidhir2k1
@179priyasoni
@Ln11211
@parammittal16

**Comment by nik132-eng** [comment]
I'm happy to do it

**Comment by sshiv5768** [comment]
🎉 Congratulations @madiedgar and team! I am ready to contribute on the coding side, can you also ping issues in the coding part.

**Comment by sneha-thyagarajan** [comment]
@madiedgar , I'm wondering if there is still a need for Hindi translation reviewers. If so, I would be happy to contribute in this one too.

**Comment by madiedgar** [comment]
@sneha-thyagarajan we actually just need one more reviewer for Python in Hindi before we can put it up for official review. (We have enough reviewers on keywords, but Python's builtin functions are lacking one review) If you wouldn't mind adding your review for both Python keywords and builtin functions that would be wonderful!!

**Comment by sneha-thyagarajan** [comment]
@madiedgar sure! I can do that

**Comment by sneha-thyagarajan** [comment]
Hi @madiedgar, I finished reviewing the Hindi translations and I've created a pull request. Please let me know if you have any changes or feedback.

**Comment by madiedgar** [comment]
This is so exciting team! We've completed this stage for Hindi's Python translations. I'll be reaching out soon to help us officially integrate Hindi into Legesher Translations! 🎉 Thank you everyone for all of your incredible work and collaboration!
